attr.go

```diff
@@ -0,0 +1,383 @@
+//Copyright 2017 GoGraphviz Authors
+//
+//Licensed under the Apache License, Version 2.0 (the "License");
+//you may not use this file except in compliance with the License.
+//You may obtain a copy of the License at
+//
+//    http)://www.apache.org/licenses/LICENSE-2.0
+//
+//Unless required by applicable law or agreed to in writing, software
+//distributed under the License is distributed on an "AS IS" BASIS,
+//WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//See the License for the specific language governing permissions and
+//limitations under the License.
+
+package gographviz
+
+import "fmt"
+
+// Attr is an attribute key
+type Attr string
+
+// NewAttr creates a new attribute key by checking whether it is a valid key
+func NewAttr(key string) (Attr, error) {
+	a, ok := validAttrs[key]
+	if !ok {
+		return Attr(""), fmt.Errorf("%s is not a valid attribute", key)
+	}
+	return a, nil
+}
+
+const (
+	DAMPING            Attr = "Damping"
+	K                  Attr = "K"
+	URL                Attr = "URL"
+	_BACKGROUND        Attr = "_background"
```

Comment:
Had this discussion with my friend the other day, and we were of different opinions. He were accustomed to ALL_CAPS from C. I had started to prefer CamelCase, as the Go project also seems to have moved in this direction recently.

For instance, from [os](https://godoc.org/os#SEEK_SET)

```go
const (
    SEEK_SET int = 0 // seek relative to the origin of the file
    SEEK_CUR int = 1 // seek relative to the current offset
    SEEK_END int = 2 // seek relative to the end
)
//  Deprecated: Use io.SeekStart, io.SeekCurrent, and io.SeekEnd.
```

The ALL_CAPS enums were deprecated (for dependency reasons), and the new enums defined in `io` were CamelCase.